masq/src/commands/financials_command.rs

```diff
@@ -58,26 +175,911 @@ impl Command for FinancialsCommand {
+                    } else {
+                        leading_zeros.0
+                            - if let Some(idx) = decimal_dot_position {
+                                if (idx - leading_zeros.0) == 0 {
+                                    1
+                                } else {
+                                    0
+                                }
+                            } else if leading_zeros.0 == str.len() {
+                                1
+                            } else {
+                                0
+                            }
+                    })
+                    .collect()
+            }
+            fn make_decision(after_care: String, before: &String) -> Option<String> {
+                if after_care.len() != before.len() {
+                    Some(after_care)
+                } else {
+                    None
+                }
+            }
+            if let Some(idx) = str.chars().position(|char| char == '.') {
+                let after_care = front_care(str, Some(idx))
+                    .chars()
+                    .rev()
+                    .skip_while(|char| *char == '0')
+                    .collect::<String>()
+                    .chars()
+                    .rev()
+                    .collect::<String>();
+                if after_care.len() != str.len() {
+                    Some(after_care)
+                } else {
+                    None
+                }
+            } else {
+                let after_care = front_care(str, None);
+                make_decision(after_care, str)
+            }
+        }
+
+        let ((time_min, time_max), (amount_min, amount_max)) = user_ranges;
+        match (
+            apply_care(time_min),
+            apply_care(time_max),
+            apply_care(amount_min),
+            apply_care(amount_max),
+        ) {
+            (None, None, None, None) => Unchanged((
+                format!("{}-{}", time_min, time_max),
+                format!("{}-{}", amount_min, amount_max),
+            )),
+            (a, b, c, d) => Changed((
+                format!(
+                    "{}-{}",
+                    a.unwrap_or_else(|| time_min.to_string()),
+                    b.unwrap_or_else(|| time_max.to_string())
+                ),
+                format!(
+                    "{}-{}",
+                    c.unwrap_or_else(|| amount_min.to_string()),
+                    d.unwrap_or_else(|| amount_max.to_string())
+                ),
+            )),
+        }
+    }
+
+    fn no_records_found<F>(stdout: &mut dyn Write, headings: &[&str], write_headings: F)
+    where
+        F: Fn(&mut dyn Write, &[&str], &[usize]),
+    {
+        let headings_widths = Self::widths_of_dynamical_headings(headings);
+        write_headings(stdout, headings, &headings_widths);
+        short_writeln!(stdout, "\nNo records found",)
+    }
+
+    fn figure_out_max_widths(records: &[&dyn WidthInfo], is_gwei: bool) -> Vec<usize> {
```

Comment:
I don't understand how this works or really what it does. You said to make a comment.